test/TestProjects/MgmtDiscriminator/src/Generated/ArmMgmtDiscriminatorModelFactory.cs

```diff
@@ -95,18 +95,18 @@ public static DeliveryRuleProperties DeliveryRuleProperties(int? order = null, D
         /// <param name="name"> The name of the condition for the delivery rule. </param>
         /// <param name="foo"> For test. </param>
         /// <returns> A new <see cref="Models.DeliveryRuleCondition"/> instance for mocking. </returns>
-        public static DeliveryRuleCondition DeliveryRuleCondition(string name = "Unknown", string foo = null)
+        public static DeliveryRuleCondition DeliveryRuleCondition(string name = null, string foo = null)
         {
-            return new UnknownDeliveryRuleCondition(name, foo, serializedAdditionalRawData: null);
+            return new UnknownDeliveryRuleCondition(name == null ? default : new MatchVariable(name), foo, serializedAdditionalRawData: null);
```

Comment:
If `name == null`, then we pass in `default` of `MatchVariable` of which the `_value` property is null: https://github.com/Azure/autorest.csharp/blob/b8f92b9a09b6b7c5fde355b8025daee99681e677/test/TestProjects/MgmtDiscriminator/src/Generated/Models/MatchVariable.cs#L16

It seems that will cause exception at runtime.

Reply:
No it will not cause runtime exception.
Default of a struct is constructed directly without calling the ctor.